src/Mvc/Mvc.Core/src/Formatters/FormatFilter.cs

```diff
@@ -171,7 +170,11 @@ public void OnResultExecuting(ResultExecutingContext context)
 
             var contentType = _options.FormatterMappings.GetMediaTypeMappingForFormat(format);
             objectResult.ContentTypes.Clear();
-            objectResult.ContentTypes.Add(contentType);
+
+            if (contentType != null)
```

Comment:
`GetMediaTypeMappingForFormat` can return a null value (for instance you do not have a mapping for a format). Prior to this, we would add a `null` valued MediaType and have it be ignored since it would fail to parse. This seems like a slightly better solution to it. I do not want to require `GetMediaTypeMappingForFormat` to return a non-null value.